src/main/java/de/tum/in/www1/artemis/service/metis/ConversationMessagingService.java

```diff
@@ -0,0 +1,232 @@
+        return createdMessage;
+    }
+
+    /**
+     * fetch posts from database by conversationId
+     *
+     * @param pageable          requested page and page size
+     * @param postContextFilter request object to fetch posts
+     * @return page of posts that match the given context
+     */
+    public Page<Post> getMessages(Pageable pageable, @Valid PostContextFilter postContextFilter) {
+        Page<Post> conversationPosts;
+        if (postContextFilter.getConversationId() != null) {
+            var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
+            var conversation = conversationService.mayInteractWithConversationElseThrow(postContextFilter.getConversationId(), requestingUser);
+            conversationPosts = conversationMessageRepository.findMessages(postContextFilter, pageable);
+            // protect sample solution, grading instructions, etc.
+            conversationPosts.stream().map(Post::getExercise).filter(Objects::nonNull).forEach(Exercise::filterSensitiveInformation);
+            setAuthorRoleOfPostings(conversationPosts.getContent());
+
+            var participantOfRequestingUser = conversationParticipantRepository.findConversationParticipantByConversationIdAndUserIdElseThrow(conversation.getId(),
+                    requestingUser.getId());
+            participantOfRequestingUser.setLastRead(ZonedDateTime.now());
+            participantOfRequestingUser.setUnreadMessagesCount(0L);
+            conversationParticipantRepository.save(participantOfRequestingUser);
+        }
+        else {
+            throw new BadRequestAlertException("A new message post cannot be associated with more than one context", METIS_POST_ENTITY_NAME, "ambiguousContext");
+        }
+        return conversationPosts;
+    }
+
+    /**
+     * Checks course, user and post validity,
+     * updates non-restricted field of the post, persists the post,
+     * and ensures that sensitive information is filtered out
+     *
+     * @param courseId    id of the course the post belongs to
+     * @param postId      id of the post to update
+     * @param messagePost post to update
+     * @return updated post that was persisted
+     */
+    public Post updateMessage(Long courseId, Long postId, Post messagePost) {
+        final User user = userRepository.getUserWithGroupsAndAuthorities();
+        // check
+        if (messagePost.getId() == null || !Objects.equals(messagePost.getId(), postId)) {
+            throw new BadRequestAlertException("Invalid id", METIS_POST_ENTITY_NAME, "idnull");
+        }
+        final Course course = preCheckUserAndCourse(user, courseId);
+
+        Post existingMessage = conversationMessageRepository.findMessagePostByIdElseThrow(postId);
+        Conversation conversation = mayUpdateOrDeleteMessageElseThrow(existingMessage, user);
+
+        // ToDo: find a cleaner way to do this instead of making the string here in the server
```

Comment:
Open todo. I agree, we should not create this string on the server but instead let the client handle the translation.

Reply:
Will be done in a follow up.